launcher/agent/agent.go

```diff
@@ -115,27 +152,53 @@ func (a *agent) Attest(ctx context.Context, opts AttestAgentOpts) ([]byte, error
 		return nil, fmt.Errorf("failed to get principal tokens: %w", err)
 	}
 
-	var buf bytes.Buffer
-	if err := a.cosCel.EncodeCEL(&buf); err != nil {
-		return nil, err
-	}
-
-	attestation, err := a.attest(challenge.Nonce, buf.Bytes())
-	if err != nil {
-		return nil, fmt.Errorf("failed to attest: %v", err)
-	}
-
 	req := verifier.VerifyAttestationRequest{
 		Challenge:      challenge,
 		GcpCredentials: principalTokens,
-		Attestation:    attestation,
 		TokenOptions: verifier.TokenOptions{
 			CustomAudience: opts.Aud,
 			CustomNonce:    opts.Nonces,
 			TokenType:      opts.TokenType,
 		},
 	}
 
+	attResult, err := a.ar.Attest(challenge.Nonce)
+	if err != nil {
+		return nil, fmt.Errorf("failed to attest: %v", err)
+	}
+
+	var cosCel bytes.Buffer
+	if err := a.cosCel.EncodeCEL(&cosCel); err != nil {
+		return nil, err
+	}
+
+	switch v := attResult.(type) {
+	case *pb.Attestation:
+		a.logger.Println("attestation through TPM quote")
+
+		v.CanonicalEventLog = cosCel.Bytes()
+		req.Attestation = v
+	case *verifier.TDCCELAttestation:
+		a.logger.Println("attestation through TDX quote")
+
+		certChain, err := internal.GetCertificateChain(a.fetchedAK.Cert(), http.DefaultClient)
+		if err != nil {
+			return nil, fmt.Errorf("failed when fetching certificate chain: %w", err)
+		}
+
+		var cosCelRtmr bytes.Buffer
+		if err := a.cosCel.EncodeCEL(&cosCelRtmr); err != nil {
+			return nil, err
+		}
```

Comment:
remove it since no callers use this variable.

Reply:
removed